src/FieldData.php

```diff
@@ -134,17 +145,30 @@ protected function entityReferenceFieldData() {
     $targetEntityTypeId = $fieldDefinition->getSetting('target_type');
     $settingName = ($targetEntityTypeId == 'paragraph') ? 'target_bundles_drag_drop' : 'target_bundles';
 
-    $targetBundles = [];
-    if (!empty($fieldDefinition->getSetting('handler_settings')[$settingName])) {
-      $targetBundles = array_map(function ($bundle) use ($targetEntityTypeId) {
-        return $this->mapping->getBundleMapping($targetEntityTypeId, $bundle);
-      }, array_keys($fieldDefinition->getSetting('handler_settings')[$settingName]));
+    $installedBundles = $this->bundleInfo->getBundleInfo($targetEntityTypeId);
+    $targetBundles = $fieldDefinition->getSetting('handler_settings')[$settingName];
```

Comment:
Just a thought.

We are using only keys from `$targetBundles` and `$installedBundles`. If we would use `array_keys` for them at the beginning, then we don't have to play around that afterwards. And also we could maybe kick out all `!empty` checks, because they are mainly gates for `NULL` values.

I don't know if something like this would work? 
```$targetBundles = array_keys($fieldDefinition->getSetting('handler_settings')[$settingName] ?: []);```